applications/sintering/include/pf-applications/sintering/postprocessors.h

```diff
@@ -1338,6 +1354,59 @@ namespace Sintering
       data_out.write_vtu_in_parallel(output, dof_handler.get_communicator());
     }
 
+    template <int dim, typename VectorType>
+    void
+    output_porosity_stats(const DoFHandler<dim> &dof_handler,
+                          const VectorType &     solution,
+                          const std::string      output)
+    {
+      const double invalid_particle_id = -1.0; // TODO
+
+      LinearAlgebra::distributed::Vector<double> particle_ids(
+        dof_handler.get_triangulation()
+          .global_active_cell_index_partitioner()
+          .lock());
+
+      // Detect pores and assign ids
+      auto [local_to_global_particle_ids, offset] = internal::detect_pores(
```

Comment:
The vector `particle_ids` still seems to be empty so how is it possible to run `detect_pores`?

Reply:
`particle_ids ` is filled inside `detect_pores()`. It was indeed confusing, I have refactored that to be more explicit.